src/main/java/frc/robot/subsystems/tank/TankSubsystem.java

```diff
@@ -131,4 +182,24 @@ public void setTankDrivePowers(double leftScale, double rightScale) {
   private double squareInput(double value) {
     return Math.copySign(value * value, value);
   }
+
+  /**
+   * Gets the estimated current position of the robot.
+   * @return the estimated position of the robot as a Pose2d.
+   */
+  public Pose2d getRobotPosition() {
+    return odometry.getEstimatedPosition();
+  }
+
+  /**
+   * Zeros the robot's position.
+   * This method zeros both the robot's translation *and* rotation.
+   */
+  public void zeroPosition() {
+    leftMain.setSelectedSensorPosition(0);
+    rightMain.setSelectedSensorPosition(0);
+
+    // https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj/estimator/DifferentialDrivePoseEstimator.html#resetPosition(edu.wpi.first.wpilibj.geometry.Pose2d,edu.wpi.first.wpilibj.geometry.Rotation2d)
+    odometry.resetPosition(new Pose2d(), Rotation2d.fromDegrees(ahrs.getAngle()));
```

Comment:
why not reset the gyro

Reply:
https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj/estimator/DifferentialDrivePoseEstimator.html#resetPosition(edu.wpi.first.wpilibj.geometry.Pose2d,edu.wpi.first.wpilibj.geometry.Rotation2d)
> The gyroscope angle does not need to be reset here on the user's robot code. The library automatically takes care of offsetting the gyro angle.

And the angle was reset to 0 during testing so it seems to work.